src/sentry/static/sentry/app/utils/integrationUtil.tsx

```diff
@@ -18,251 +18,174 @@ import {
-    role: org?.role,
-    custom_referrer,
-    ...features,
-    ...analyticsParams,
-  };
+export type AnalyticsKey = keyof EventParameters;
+
+//define the event key to event name mappings
+const eventNameMap: Record<AnalyticsKey, string> = {
+  'integrations.install_modal_opened': 'Integrations: Install Modal Opened',
+  'integrations.integration_viewed': 'Integrations: Integration Viewed',
+  'integrations.installation_start': 'Integrations: Installation Start',
+  'integrations.installation_complete': 'Integrations: Installation Complete',
+  'integrations.uninstall_clicked': 'Integrations: Uninstall Clicked',
+  'integrations.uninstall_completed': 'Integrations: Uninstall Completed',
+  'integrations.enabled': 'Integrations: Enabled',
+  'integrations.disabled': 'Integrations: Disabled',
+  'integrations.config_saved': 'Integrations: Config Saved',
+  'integrations.integration_tab_clicked': 'Integrations: Integration Tab Clicked',
+  'integrations.plugin_add_to_project_clicked':
+    'Integrations: Plugin Add to Project Clicked',
+  'integrations.resolve_now_clicked': 'Integrations: Resolve Now Clicked',
+  'integrations.request_install': 'Integrations: Request Install',
+  'integrations.code_mappings_viewed': 'Integrations: Code Mappings Viewed',
+  'integrations.details_viewed': 'Integrations: Details Viewed',
+  'integrations.index_viewed': 'Integrations: Index Page Viewed',
+  'integrations.directory_item_searched': 'Integrations: Directory Item Searched',
+  'integrations.directory_category_selected': 'Integrations: Directory Category Selected',
+  'integrations.stacktrace_start_setup': 'Integrations: Stacktrace Start Setup',
+  'integrations.stacktrace_submit_config': 'Integrations: Stacktrace Submit Config',
+  'integrations.stacktrace_complete_setup': 'Integrations: Stacktrace Complete Setup',
+  'integrations.stacktrace_manual_option_clicked':
+    'Integrations: Stacktrace Manual Option Clicked',
+  'integrations.stacktrace_link_clicked': 'Integrations: Stacktrace Link Clicked',
+  'integrations.reconfigure_stacktrace_setup':
+    'Integrations: Reconfigure Stacktrace Setup',
+  'integrations.stacktrace_docs_clicked': 'Integrations: Stacktrace Docs Clicked',
+
+  'integrations.serverless_functions_viewed': 'Integrations: Serverless Functions Viewed',
+  'integrations.installation_input_value_changed':
+    'Integrations: Installation Input Value Changed',
+  'integrations.serverless_function_action': 'Integrations: Serverless Function Action',
+  'integrations.cloudformation_link_clicked': 'Integrations: CloudFormation Link Clicked',
+  'integrations.switch_manual_sdk_setup': 'Integrations: Switch Manual SDK Setup',
+};
 
-  //add the integration_status to the type of params so TS doesn't complain about what we do below
-  const fullParams: typeof params & {
-    integration_status?: SentryAppStatus;
-  } = params;
+//hook to lazyily generate eventNameMap
+export const getEventNameMap = () => eventNameMap;
```

Comment:
Do we actually need this as a function?

Reply:
@EvanPurkhiser I think the cross-importing of the two modules was breaking things. Because when I imported `eventNameMap` directly, it was coming out as an empty object. That's actually why I added the check you made a comment about: https://github.com/getsentry/sentry/pull/23668/files/fdd88cee26f68b45414d6e9d7b413c124631608b#r572519618